### PR TITLE
Updates to ATOM

### PR DESCRIPTION
**Pull request by williamdjones**
I'm adding all of the scripts (bash and python) that I used to work with the chembl and enamine datasets for this project.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,14 @@ data.prototext*
...
 # tmp/**/* 
 build
+spack_environments/users/
+
+# can ignore experiment/data directories
+applications/ATOM/charvae_data
+applications/ATOM/charvae_experiments
+
+# can ignore the moses repo
+applications/ATOM/moses/
```
Comment (timmoon10) [inline .gitignore:23]: Not a big deal, but I'd recommend putting this inside a `.gitignore` file within `applications/ATOM`. That would keep it more self-contained.

**Comment by samadejacobs** [comment]
This is now ready for review. The three files to focus on are (1) train_atom_char_rnn.py (2) dataset.py (3) data_config.json, I believe the other files should not go into LBANN repo (be deleted). Largely, this refactor will let us support multiple dataset (zinc,cheml, enamine etc), with zinc as default dataset (as it is right now before this PR). Lastly, I see there is a long list of reviewers, IMHO, @timmoon10 is competent enough to review and merge this but others are free to provide additional reviews bear in mind that this application is being actively used (and urgent, kinda!)

File: applications/ATOM/train_atom_char_rnn.py [hunk 2]
```diff
@@ -40,133 +84,194 @@ def construct_model():
...
 
     # Training set data reader
     data_reader = message.reader.add()
-    data_reader.name = 'python'
-    data_reader.role = 'train'
+    data_reader.name = "python"
+    data_reader.role = "train"
     data_reader.shuffle = True
-    data_reader.percent_of_data_to_use = 1.0
+    data_reader.percent_of_data_to_use = 0.001
```
Comment (timmoon10) [inline applications/ATOM/train_atom_char_rnn.py:207]: Is it intentional to use 0.1% of the data?

Comment (samadejacobs) [reply applications/ATOM/train_atom_char_rnn.py]: I changed it back to 1.0

File: .gitignore (hunk 1 above)
Comment (benson31) [reply .gitignore:23]: Seconded.

Comment (samadejacobs) [reply .gitignore]: Done

File: .gitmodules [hunk 3]
```diff
@@ -6,3 +6,6 @@
 	path = applications/graph/largescale_node2vec
 	url = https://lc.llnl.gov/bitbucket/scm/havoq/largescale_node2vec.git
 	ignore = dirty
+[submodule "applications/ATOM/moses"]
+	path = applications/ATOM/moses
+	url = git@github.com:williamdjones/moses.git
```
Comment (bvanessen) [inline .gitmodules:11]: Should this point to Derek's github or the published one?

Comment (samadejacobs) [reply .gitmodules]: Fixed, points to my fork which at this time has all the necessary scripts needed for ATOM specific dataset, these scripts were contributed by Derek.

**Pull request merged**